src/test/java/org/rulez/demokracia/pdengine/ChoiceModifyValidationTest.java

```diff
@@ -0,0 +1,90 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.tested_behaviour;
+import org.rulez.demokracia.pdengine.annotations.tested_feature;
+import org.rulez.demokracia.pdengine.annotations.tested_operation;
+import org.rulez.demokracia.pdengine.dataobjects.ChoiceEntity;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultVoteRegistry;
+
+public class ChoiceModifyValidationTest extends CreatedDefaultVoteRegistry {
+	
+	public String voteId, choiceId, adminKey, choice;
+	
+	public void setUp() throws ReportedException {
+		super.setUp();
+		
+		voteId = adminInfo.voteId;
+		choiceId = voteManager.addChoice(adminInfo.adminKey, adminInfo.voteId, "choice1", "user");
+		adminKey = adminInfo.adminKey;
+		choice = "New choice name";
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_voteId_is_rejected() throws ReportedException {
+		String invalidVoteId = "invalidVoteId";
+		
+		assertThrows(
+				() -> voteManager.modifyChoice(invalidVoteId, choiceId, adminKey, choice)
+			).assertMessageIs(String.format("illegal voteId: %s",invalidVoteId));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_choiceId_is_rejected() throws ReportedException {
+		String invalidChoiceId = "invalidChoiceId";
+		
+		assertThrows(
+				() -> voteManager.modifyChoice(voteId, invalidChoiceId, adminKey, choice)
+			).assertMessageIs(String.format("Illegal choiceId: %s",invalidChoiceId));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_adminKey_is_rejected() throws ReportedException {
+		String invalidAdminKey = "invalidAdminKey";
+		
+		assertThrows(
+				() -> voteManager.modifyChoice(voteId, choiceId, invalidAdminKey, choice)
+			).assertMessageIs(String.format("Illegal adminKey: %s",invalidAdminKey));
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void ok() throws ReportedException {
+		voteManager.modifyChoice(voteId, choiceId, adminKey, choice);
+		
+		ChoiceEntity choiceEntity = voteManager.getChoice(voteId, choiceId);
+		assertEquals(choiceEntity.name, choice);
+	}
+	
+	
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void ok_with_ballot() throws ReportedException {
```

Comment:
the test case name is not verbose enough, please elaborate
the test case belongs to another behaviour: https://adadocs.demokracia.rulez.org/PDEngine/edemo/master/index.html#c49a56ca-86b8-459c-9848-a42d73666dcd
And the behaviour should have thrown a ReportedException instead of returning OK.

Reply:
So I would like to test the modifyChoice method with not empty votes ballots.
I updated the test name.